### PR TITLE
fix build when not building F# on CoreCLR

### PR DESCRIPTION
**Pull request by dsyme**

### CODE REVIEW

File: src/FSharpSource.targets [hunk 1]
```diff
@@ -737,13 +737,24 @@
...
   <Choose>
     <When Condition="'$(BuildWith)' == 'LKG'">
-      <PropertyGroup>
-         <FSharpTargetsPath>..\lkg\FSharp-$(LkgVersion)\bin\Microsoft.FSharp.Targets</FSharpTargetsPath>
-         <FscToolPath>$(FSharpSourcesRoot)\..\lkg\bin\Debug\dnxcore50\win7-x64\publish</FscToolPath>
-         <FSCoreLKGPath>$(FSharpSourcesRoot)\..\lkg\FSharp-$(LkgVersion)\bin\FSharp.Core.dll</FSCoreLKGPath>
-      </PropertyGroup>
-    </When>
-    <When Condition="'$(TargetFramework)'=='portable47' OR '$(TargetFramework)'=='portable7' OR '$(TargetFramework)'=='portable78' OR '$(TargetFramework)'=='portable259' OR '$(TargetFramework)'=='coreclr'">
+		<PropertyGroup>
```
Comment (KevinRansom) [inline src/FSharpSource.targets]: Don, the idea is that we always use the coreclr version of the lkg for building the proto-compiler.  My next step, and I have a branch where this works except for the build task is to use the coreclr lkg to build a coreclr version of the proto compiler.  These are all steps towards making the whole repo ... including the windows bits, build cross platform.

**Comment by KevinRansom** [comment]
I have a fix for this:  https://github.com/Microsoft/visualfsharp/pull/1053

**Comment by dsyme** [comment]
@KevinRansom OK. But please make it simple and easy to work on the compiler and visual studio tools without _having_ to download CoreCLR.  I would much prefer it if Proto was only run with CoreCLR when coreclr is included as a target in the build specification.  See also my comment [here](https://github.com/Microsoft/visualfsharp/pull/1053/files#r59123300)

**Comment by dsyme** [comment]
Hi Kevin, I'll reopen, so I can get it green and use it myself, but won't merge.  

(#1053 doesn't address the issue of downloading CoreCLR when doing compiler development)

**Comment by KevinRansom** [comment]
Please don’t your analysis is almost completely wrong.

**Comment by dsyme** [comment]
OK, I'll wait for #1053

**Comment by KevinRansom** [comment]
Thank you

**Comment by dsyme** [comment]
BTW I see a bunch of assemblies being resolved from the GAC when building the Proto:
```
..\..\..\lkg\bin\Debug\dnxcore50\win7-x64\publish\fsc.exe ....

-r:C:\windows\Microsoft.Net\assem
 bly\GAC_MSIL\Microsoft.Build\v4.0_14.0.0.0__b03f5f7f11d50a3a\Microsoft
 .Build.dll 

-r:C:\windows\Microsoft.Net\assembly\GAC_MSIL\Microsoft.Bui
 ld.Engine\v4.0_14.0.0.0__b03f5f7f11d50a3a\Microsoft.Build.Engine.dll 

-
 r:C:\windows\Microsoft.Net\assembly\GAC_MSIL\Microsoft.Build.Framework
 \v4.0_14.0.0.0__b03f5f7f11d50a3a\Microsoft.Build.Framework.dll 

-r:C:\w
 indows\Microsoft.Net\assembly\GAC_MSIL\Microsoft.Build.Tasks.core\v4.0
 _14.0.0.0__b03f5f7f11d50a3a\Microsoft.Build.Tasks.core.dll -r:C:\windo
 ws\Microsoft.Net\assembly\GAC_MSIL\Microsoft.Build.Utilities.core\v4.0
 _14.0.0.0__b03f5f7f11d50a3a\Microsoft.Build.Utilities.core.dll 

-r:C:\windows\Microsoft.Net\assembly\GAC_MSIL\System.Collections.
 Immutable\v4.0_1.1.37.0__b03f5f7f11d50a3a\System.Collections.Immutable
 .dll 


```

**Comment by dsyme** [comment]
Also this warning: 

capture72

**Comment by dsyme** [comment]
Just some stats - using the CoreCLR proto added about 8 minutes of download for me before getting to the first stage of compilation.

(I understand there are big advantages to using a CoreCLR proto and that it should soon allow a Linux development experience for the repo.  I'm just not feeling we should _require_ it just yet, e.g. for people who just want to work on the VS tools ).

**Comment by KevinRansom** [comment]
There are plenty of optimizations we will be able to do, to ensure that half the internet is not downloaded.  When building the compiler, we just aren’t doing them yet, part of that is the tools aren’t finished by a mile, the other is we are still learning how to use them.  There are many other things that are coming that will fix it, we just have to figure out what they are and how to benefit from them.

What we have is not terrible, and there is plenty we can do to improve things.  I’m guessing we will.

Kevin

**Comment by KevinRansom** [comment]
resolved by #1055

**Comment by smoothdeveloper** [comment]
Hello, I'm trying to run `build.cmd` (or `build.cmd build` to just compile) but face this message:

Also, since I'm on a slow link, how can I make sure I'm not facing the "download gigs of dotnet core" situation?

Thanks for your help!

```
C:\dev\src\projects\github.com\visualfsharp>build                                                                                                                          
Parse argument build                                                                                                                                                       
                                                                                                                                                                           
Build/Tests configuration:                                                                                                                                                 
                                                                                                                                                                           
BUILD_NET40=1                                                                                                                                                              
BUILD_CORECLR=0                                                                                                                                                            
BUILD_PORTABLE=1                                                                                                                                                           
BUILD_VS=1                                                                                                                                                                 
BUILD_FSHARP_DATA_TYPEPROVIDERS=0                                                                                                                                          
BUILD_CONFIG=release                                                                                                                                                       
BUILD_CONFIG_LOWERCASE=release                                                                                                                                             
                                                                                                                                                                           
TEST_COMPILERUNIT=0                                                                                                                                                        
TEST_PORTABLE_COREUNIT=0                                                                                                                                                   
TEST_VS=0                                                                                                                                                                  
TEST_FSHARP_SUITE=0                                                                                                                                                        
TEST_FSHARPQA_SUITE=0                                                                                                                                                      
TEST_TAGS=                                                                                                                                                                 
                                                                                                                                                                           
'gacutil' is not recognized as an internal or external command,                                                                                                            
operable program or batch file.                                                                                                                                            
                                                                                                                                                                           
C:\dev\src\projects\github.com\visualfsharp>if not '' == '' goto vsversionset                                                                                              
                                                                                                                                                                           
C:\dev\src\projects\github.com\visualfsharp>if exist "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\Tools\..\ide\devenv.exe" set VisualStudioVersion=14.0    
                                                                                                                                                                           
C:\dev\src\projects\github.com\visualfsharp>if exist "C:\Program Files (x86)\Microsoft Visual Studio 14.0\common7\ide\devenv.exe" set VisualStudioVersion=14.0             
                                                                                                                                                                           
C:\dev\src\projects\github.com\visualfsharp>if exist "C:\Program Files\Microsoft Visual Studio 14.0\common7\ide\devenv.exe" set VisualStudioVersion=14.0                   
                                                                                                                                                                           
C:\dev\src\projects\github.com\visualfsharp>if not '14.0' == '' goto vsversionset                                                                                          
                                                                                                                                                                           
C:\dev\src\projects\github.com\visualfsharp>if '14.0' == '' echo Error: Could not find an installation of Visual Studio   && goto :failure                                 
                                                                                                                                                                           
C:\dev\src\projects\github.com\visualfsharp>if exist "C:\Program Files (x86)\MSBuild\14.0\Bin\MSBuild.exe" set _msbuildexe="C:\Program Files (x86)\MSBuild\14.0\Bin\MSBuild
.exe"                                                                                                                                                                      
                                                                                                                                                                           
C:\dev\src\projects\github.com\visualfsharp>if exist "C:\Program Files\MSBuild\14.0\Bin\MSBuild.exe" set _msbuildexe="C:\Program Files\MSBuild\14.0\Bin\MSBuild.exe"       
                                                                                                                                                                           
C:\dev\src\projects\github.com\visualfsharp>if not exist "C:\Program Files (x86)\MSBuild\14.0\Bin\MSBuild.exe" echo Error: Could not find MSBuild.exe.   && goto :failure  
                                                                                                                                                                           
C:\dev\src\projects\github.com\visualfsharp>if defined APPVEYOR (                                                                                                          
rem See <http://www.appveyor.com/docs/build-phase>                                                                                                                         
 if exist "C:\Program Files\AppVeyor\BuildAgent\Appveyor.MSBuildLogger.dll" (                                                                                              
rem HACK HACK HACK                                                                                                                                                         
 set _msbuildexe="C:\Program Files (x86)\MSBuild\14.0\Bin\MSBuild.exe" /logger:"C:\Program Files\AppVeyor\BuildAgent\Appveyor.MSBuildLogger.dll"                           
)                                                                                                                                                                          
)                                                                                                                                                                          
                                                                                                                                                                           
C:\dev\src\projects\github.com\visualfsharp>set msbuildflags=/maxcpucount                                                                                                  
                                                                                                                                                                           
C:\dev\src\projects\github.com\visualfsharp>set _ngenexe="C:\Windows\Microsoft.NET\Framework\v4.0.30319\ngen.exe"                                                          
                                                                                                                                                                           
C:\dev\src\projects\github.com\visualfsharp>if not exist "C:\Windows\Microsoft.NET\Framework\v4.0.30319\ngen.exe" echo Error: Could not find ngen.exe.   && goto :failure  
                                                                                                                                                                           
C:\dev\src\projects\github.com\visualfsharp>if 'true' == 'true' (                                                                                                          
"C:\Windows\Microsoft.NET\Framework\v4.0.30319\ngen.exe" install .\.nuget\NuGet.exe                                                                                        
 .\.nuget\NuGet.exe restore packages.config -PackagesDirectory packages -ConfigFile .nuget\nuget.config                                                                    
                                                                                                                                                                           
)                                                                                                                                                                          
Microsoft (R) CLR Native Image Generator - Version 4.6.1055.0                                                                                                              
Copyright (c) Microsoft Corporation.  All rights reserved.                                                                                                                 
Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED))                                                                                                    
Administrator permissions are needed to use the selected options.  Use an administrator command prompt to complete these tasks.                                            
All packages listed in packages.config are already installed.                                                                                                              
                                                                                                                                                                           
C:\dev\src\projects\github.com\visualfsharp>rem Always do this it installs the dotnet cli and publishes the LKG                                                            
                                                                                                                                                                           
C:\dev\src\projects\github.com\visualfsharp>rem ===============================================================                                                            
                                                                                                                                                                           
C:\dev\src\projects\github.com\visualfsharp>set DOTNET_HOME=.\packages\dotnet                                                                                              
                                                                                                                                                                           
C:\dev\src\projects\github.com\visualfsharp>set _dotnetexe=.\packages\dotnet\dotnet.exe                                                                                    
                                                                                                                                                                           
C:\dev\src\projects\github.com\visualfsharp>rem check to see if the dotnet cli tool exists                                                                                 
                                                                                                                                                                           
C:\dev\src\projects\github.com\visualfsharp>if not exist .\packages\dotnet\dotnet.exe (                                                                                    
echo Could not find .\packages\dotnet\dotnet.exe. Do zipfile install                                                                                                       
 if not exist packages (md packages  )                                                                                                                                     
 if exist packages\dotnet (rd packages\dotnet /s /q  )                                                                                                                     
 powershell.exe -executionpolicy unrestricted -command .\scripts\install-dotnetcli.ps1 https://dotnetcli.blob.core.windows.net/dotnet/beta/Binaries/Latest/dotnet-dev-win-x
64.latest.zip packages                                                                                                                                                     
                                                                                                                                                                           
 pushd .\lkg   & ..\.\packages\dotnet\dotnet.exe restore   & popd                                                                                                          
                                                                                                                                                                           
 pushd .\lkg   & ..\.\packages\dotnet\dotnet.exe publish project.json   & popd                                                                                             
                                                                                                                                                                           
 rem rename fsc and coreconsole to allow fsc.exe to to start compiler                                                                                                      
 pushd .\lkg\bin\debug\dnxcore50\win7-x64\publish                                                                                                                          
 ren fsc.exe fsc.dll                                                                                                                                                       
 copy corehost.exe fsc.exe                                                                                                                                                 
 popd                                                                                                                                                                      
 rem rename fsi and coreconsole to allow fsi.exe to to start interative                                                                                                    
 pushd .\lkg\bin\debug\dnxcore50\win7-x64\publish                                                                                                                          
 ren fsi.exe fsi.dll                                                                                                                                                       
 copy corehost.exe fsi.exe                                                                                                                                                 
 popd                                                                                                                                                                      
)                                                                                                                                                                          
Could not find .\packages\dotnet\dotnet.exe. Do zipfile install                                                                                                            
The term 'Invoke-WebRequest' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was inclu 
ded, verify that the path is correct and try again.                                                                                                                        
At C:\dev\src\projects\github.com\visualfsharp\scripts\install-dotnetcli.ps1:9 char:18                                                                                     
+ Invoke-WebRequest <<<<  -UseBasicParsing $clipath -OutFile $BackUpPath                                                                                                   
    + CategoryInfo          : ObjectNotFound: (Invoke-WebRequest:String) [], CommandNotFoundException                                                                      
    + FullyQualifiedErrorId : CommandNotFoundException                                                                                                                     
                                                                                                                                                                           
Add-Type : Cannot add type. The assembly 'system.io.compression.filesystem' could not be found.                                                                            
At C:\dev\src\projects\github.com\visualfsharp\scripts\install-dotnetcli.ps1:10 char:9                                                                                     
+ Add-Type <<<<  -assembly "system.io.compression.filesystem"                                                                                                              
    + CategoryInfo          : ObjectNotFound: (system.io.compression.filesystem:String) [Add-Type], Exception                                                              
    + FullyQualifiedErrorId : ASSEMBLY_NOT_FOUND,Microsoft.PowerShell.Commands.AddTypeCommand                                                                              
                                                                                                                                                                           
Add-Type : Cannot add type. One or more required assemblies are missing.                                                                                                   
At C:\dev\src\projects\github.com\visualfsharp\scripts\install-dotnetcli.ps1:10 char:9                                                                                     
+ Add-Type <<<<  -assembly "system.io.compression.filesystem"                                                                                                              
    + CategoryInfo          : InvalidData: (:) [Add-Type], InvalidOperationException                                                                                       
    + FullyQualifiedErrorId : ASSEMBLY_LOAD_ERRORS,Microsoft.PowerShell.Commands.AddTypeCommand                                                                            
                                                                                                                                                                           
Unable to find type [io.compression.zipfile]: make sure that the assembly containing this type is loaded.                                                                  
At C:\dev\src\projects\github.com\visualfsharp\scripts\install-dotnetcli.ps1:11 char:25                                                                                    
+ [io.compression.zipfile] <<<< ::ExtractToDirectory($BackUpPath, $destination)                                                                                            
    + CategoryInfo          : InvalidOperation: (io.compression.zipfile:String) [], RuntimeException                                                                       
    + FullyQualifiedErrorId : TypeNotFound                                                                                                                                 
                                                                                                                                                                           
The system cannot find the path specified.                                                                                                                                 
Error: dotnet restore failed                                                                                                                                               
```

**Comment by dsyme** [comment]
@smoothdeveloper The download size is about 100MB we think/ It won't bee a problem.  I was just on a slow connection.

The error above stems from this:

> The term 'Invoke-WebRequest' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was inclu  ded, verify that the path is correct and try again.      

Not sure what the problem is. Are you running from an administrator shell?

**Comment by smoothdeveloper** [comment]
@dsyme 100mb seems reasonable (although my link is around 60kb/s) if it stays cached on my machine / not churning too much.

I just tried under administrator shell, and get same error.

**Comment by smoothdeveloper** [comment]
do I need anything special WRT powershell? I'm on windows 7 and don't have much powershell foo.

Only powershell stuff in my start menu seems to be default system install, located in `%SystemRoot%\system32\WindowsPowerShell\v1.0\powershell.exe` and the matching sysWOW64 folder.

**Comment by dsyme** [comment]
@smoothdeveloper I don't know I'm afraid. Maybe @KevinRansom knows.  (It would be great to get rid of all the PS scripts - they aren't cross-platform anyway, and we have F# available to do the scripting!)

**Comment by smoothdeveloper** [comment]
@dsyme: .fsx scripts would be great replacement over powershell for those illiterate in powershell scripting (I assume maintainers of this repo know F# scripting better in general).

Also using FAKE & paket (not sure how confident fsharp maintainers are to push dogfooding there) would make sense IMHO.

**Comment by CumpsD** [comment]
Having the visualfsharp repo use Paket would be awesome, would help with doing coreclr support in something other than MS owned repos, as well as a strong signal in believing in F# and dogfooding it

**Comment by KevinRansom** [comment]
F# scripts are fine.

**Comment by smoothdeveloper** [comment]
@KevinRansom, thanks, what about FAKE?

I think target / dependency graph support would be good.

Can we get some pointers what we would like to see tried first if someone would like to sketch replacing one of the simplest pieces?

**Comment by dsyme** [comment]
@smoothdeveloper We've discussed FAKE before - there are issues and PRs on it, search around in GitHub's search.  I think we've decided it's not going to happen, build requirements in this repo are pretty specialized esp. for portable, CoreCLR, VSIX, MicroBuild etc.

**Comment by forki** [comment]
Yes there were issues and pull requests, but AFAIK we never learned why
FAKE didn't make it in. I mean it's fine and I don't want to try another
time, but at least I never learned why.

**Comment by smoothdeveloper** [comment]
@dsyme would you recommend custom parsing of arguments, managing task dependencies etc?

I understand the requirements for the master scripts are quite involving, but at the same time for a layman F# contributor, having a fake script with targets that are more discoverable etc. shouldn't hinder much the process of replacing some .bat / .ps files.

**Comment by dsyme** [comment]
@smoothdeveloper I love FAKE and use it all the time in pretty much every other project. But this one it's just not going to happen for a loooong time.  I'm sure we could make it happen, but there's sooooo much else to do and FAKE may or may not commute with other stuff going on.  And the churn/re-education required don't feel like they will justify it.  I FAKE a lot, but not here.

For example, Microsoft use this MSBuild thing called MicroBuild as part of their internal build. Kevin and Omar have just added MicroBuild support to this repo so we can eventually do fully public builds of the actual Visual F# Tools installers (the VSIX you get from this repo is not the _actual_ installer).  I have no idea if FAKE needs special support for MicroBuild, and even if we did solve that there may be a host of other issues.  There are just other priorities for now.

Thanks to @enricosada  we _have_ managed to remove a swathe of .bat and .pl files used for testing, replaced by nice .fsx scripts.  Almost.  We're still not quite done with that. 

@smoothdeveloper Agreed we can certainly make the experience nicer for F# developers.  We tidied up ``VisualFSharp.sln`` and ``vsintegration\...`` as part of that.  There are lots of conflicting demands on this repo - especially now CoreCLR support is in - and it's hard to land simplicity.

**Pull request closed without merging**